driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -70,25 +85,78 @@ public static BooleanExpression of(final boolean of) {
     public static IntegerExpression of(final int of) {
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonInt32(of)));
     }
+
+    public static ArrayExpression<IntegerExpression> ofIntegerArray(final int... array) {
+        List<BsonValue> list = Arrays.stream(array)
+                .mapToObj(BsonInt32::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(list)));
+    }
+
     public static IntegerExpression of(final long of) {
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonInt64(of)));
     }
+
+    public static ArrayExpression<IntegerExpression> ofIntegerArray(final long... array) {
+        List<BsonValue> list = Arrays.stream(array)
+                .mapToObj(BsonInt64::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(list)));
+    }
+
     public static NumberExpression of(final double of) {
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonDouble(of)));
     }
+
+    public static ArrayExpression<IntegerExpression> ofNumberArray(final double... array) {
+        List<BsonValue> list = Arrays.stream(array)
+                .mapToObj(BsonDouble::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(list)));
+    }
+
     public static NumberExpression of(@NonNull final BigDecimal of) {
         Assertions.notNull("BigDecimal", of);
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonDecimal128(new Decimal128(of))));
     }
+
+    public static ArrayExpression<IntegerExpression> ofNumberArray(final BigDecimal... array) {
```

Comment:
Placeholder: I proposed in an earlier PR to not support BigDecimal directly.

Reply:
Propagated